pytorch_lightning/trainer/trainer.py

```diff
@@ -576,6 +577,10 @@ def fit(
         # only on proc 0 because no spawn has happened yet
         model.prepare_data()
 
+        self.mode = TrainerMode.TRAINING
+        return self._fit(model)
+
+    def _fit(self, model):
```

Comment:
Is there a better name for this to distinguish it from the public fit? Or otherwise can you just combine the functions?

Reply:
the using _ to make it protected is common case :]
what kind of combination do you he in mind?

Reply:
Yeah I understand the use of _ I just feel like if there's two functions that are different but named the same thing then it means one of them isn't named correctly. Maybe it's a bit pedantic though - happy keeping it how it is if you prefer.

Reply:
i agree with @tullie - is there a reason to split this into two functions?

Reply:
It was split into two because at the very beginning there wasn't `self.mode = TrainerMode.TRAINING` at the end of validation and testing so `.fit()` will set `self.mode = TrainerMode.TRAINING`. But since there is now, it should be fine to only use one function.